### DISCUSSION TITLE
How to access the fields of an instance in its validator?

### DISCUSSION THREAD

**Discussion by will-afs** [pydantic/pydantic#4536] [Question]
I am making a BaseModel, which field assignment should be validated. This validation should be performed in function of the current value of the field in the instance.

```Python
class SomeModel(BaseModel):
    field_to_introspect : int = 10
    
    @validator("field_to_introspect")
    @classmethod
    def ensure_is_lower_than_before(
        cls,
        eventually_new_value: int,
    ) -> int:
        if <the_instance>.field_to_introspect >= eventually_new_value:
            return <the_instance>.field_to_introspect
        else:
            return eventually_new_value
```
            
Any idea how to access the instance and its fields in a validator?

Thanks

**Comment by samuelcolvin** [comment]
In v2 you'll be able to pass a context argument to models and validators.

In v1 you should either:
* Defer that logic until after validation
* Define the model inside a function where you have the value
* Or, use contextvars
